### ISSUE TITLE
Detecting Live vs VOD mode

### ISSUE THREAD

**Issue by jfairley** [clappr/clappr#488]
I'm building custom controls for my player, and I want to implement my own "back to live" button. My stream may be Live or VOD, so I need to detect the playback type at runtime.

With some digging, I've found a way to get this information, but it's buried pretty deeply in the player. (see code snippet)

```javascript
        this.player = new window.Clappr.Player({
            source: this.buildSourcePath(this.props.directory),
            chromeless: true,
            useDvrControls: true,
            parentId: '#player'
        });

// some code ...

        let playbackType = this.player.core.mediaControl.container.getPlaybackType();
        switch (playbackType) {
            case 'live':
                // show live button
                break;
            case 'vod':
                // hide live button
                break;
        }
```

Is there a better way to detect playback type?

**Comment by tarkanlar** [comment]
I was planning to write a plugin for this. is there any progress on this?

**Comment by towerz** [comment]
@jfairley I just pushed some small changes to the core object to allow getting the current container and checking the playback type:
```javascript
> player.core.getPlaybackType()
< "vod"
> player.core.getCurrentContainer()
< Container {uniqueId: "o6", cid: "c3", $el: X[1], el: div.container.pointer-enabled, currentTime: 0…}
```

I think this will help you to avoid going too deep to find out whether you are playing live/vod content.

Please let me know if you have any progress in this or if you have any doubts.

**Comment by jfairley** [comment]
That is a much nicer call. Thanks.

I bundle the clappr player from npm. Any idea when this will go into a release?

**Comment by towerz** [comment]
I think we'll release a new version by tomorrow, just need to make a few more adjustments

**Comment by jfairley** [comment]
:+1:

**Comment by towerz** [comment]
Just released a new version (0.1.6) with these changes. Please let me know if you have any questions.

**Comment by hsa1280** [comment]
Hello, I used player.core.getPlaybackType() to detect live mode or VOD mode, the function always returns 'vod' for either live stream mode or VOD mode. Seems like this functions can't detect the right playing mode, please take a look. Thanks.
